### ISSUE TITLE
BUG: The exception is arisen when `apply` takes functions with equal names

### ISSUE THREAD

**Issue by YarShev** [pandas-dev/pandas#36036]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import pandas
pdf = pandas.DataFrame([[4, 9]] * 3, columns=['A', 'B'])
pdf.apply(["sum", "sum"])
SpecificationError: Function names must be unique if there is no new column names assigned
```

#### Problem description

Could anyone please explain? Is it normal behavior?

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
pandas           : 1.1.1
numpy            : 1.18.4
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 41.2.0
Cython           : None
pytest           : 5.4.2
hypothesis       : None
sphinx           : None
blosc            : None
feather          : 0.4.1
xlsxwriter       : None
lxml.etree       : 4.5.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.14.0
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : None
fsspec           : 0.7.3
fastparquet      : None
gcsfs            : None
matplotlib       : 3.2.1
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : 0.16.0
pytables         : None
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.4.1
sqlalchemy       : 1.3.17
tables           : 3.6.1
tabulate         : None
xarray           : 0.15.1
xlrd             : 1.2.0
xlwt             : None
numba            : None

</details>

**Comment by jbrockmendel** [comment]
The relevant part of the exception message is "Function names must be unique", which seems pretty clear.  The tail part of the message could be clearer.

**Comment by asishm** [comment]
Seems like a regression. It worked it 1.0.5 and I don't see anything in the release notes after that that address this.

```python
>>> import pandas as pd
>>> pdf = pd.DataFrame([[4, 9]] * 3, columns=['A', 'B'])
>>> pdf.apply(["sum", "sum"])
      A   B
sum  12  27
sum  12  27
>>> pd.__version__
'1.0.5'
```

git bisect returns

0534b004f5dd9cb38082a2d1e3e121b211a3373b is the first bad commit
commit 0534b004f5dd9cb38082a2d1e3e121b211a3373b
Author: Kaiqi Dong <kaiqi@kth.se>
Date:   Fri Jul 10 21:16:27 2020 +0200

    ENH: Implement Keyword Aggregation for DataFrame.agg and Series.agg (#29116)

**Comment by YarShev** [comment]
@asishm , @jbrockmendel , yes, it worked with version 1.0.5. If it is a regression, will it be fixed?

**Comment by jbrockmendel** [comment]
@charlesdong1991 thoughts on this?

**Comment by charlesdong1991** [comment]
emm, okay, it was indeed brought in to avoid having identical rows when not using named aggregation, but probably not very ideal to put this way.

can be considered as a regression indeed, I can take a closer look and draw up a PR to address it.

thanks @YarShev @asishm

**Comment by YarShev** [comment]
@charlesdong1991 , okay, thanks! Will be waiting your PR.
